enhancements/compact-clusters.md

```diff
@@ -0,0 +1,286 @@
+and since we prefer to test as our users will run, our CI environments should test
+that way as well.
+
+### Goals
+
+* Ensure 3-node clusters work correctly in cloud and metal enviroments in UPI and IPI
+* Enforce guard rails via testing that prevent regressions in 3-node environments
+* Describe the prerequisites to support single-node clusters in a future release
+* Clarify the supported behavior for single-node today
+* Identify future improvements that may be required for more resource constrained environments
+* Incentivize end users to leverage our dynamic compute and autoscaling capabilities in cloud environments
+
+### Non-Goals
+
+* Supporting any topology other than three masters for production clusters at this time
+
+## Proposal
+
+### User Stories
+
+#### Story 1
+
+As a user of OpenShift, I should be able to install a 3-node cluster (no workers)
+on cloud environments via the IPI or UPI installation paths and have a fully conformant
+OpenShift cluster (all features enabled).
+
+#### Story 2
+
+As a developer of OpenShift, I should be able to run smaller clusters that are more
+resource efficient for iterative testing while still being conformant.
+
+#### Story 3
+
+As a developer of OpenShift, I should be exposed to bugs and failures caused by
+assumptions about cluster size or the topology of cluster early, so that my feature
+additions do not break small cluster topologies.
+
+#### Story 4
+
+As an admin of an OpenShift cluster deployed onto 3-nodes, I should be able to easily
+transition to a larger cluster and make the control plane unschedulable.
+
+### Risks and Mitigations
+
+## Design Details
+
+### Enabling three-node clusters in the cloud
+
+OpenShift 4.3 should pass a full e2e run on a 3-node cluster in the cloud.
+
+The primary limitation blocking this in 4.2 and earlier was a limitation in
+Kubernetes that prevented service load balancers from targeting the master nodes
+when the node-role for masters was set. This was identified as a bug and in
+Kube 1.16 we [clarified that this should be fixed by moving the functionality to
+explicit labels](https://github.com/kubernetes/enhancements/issues/1143) and
+[added two new alpha labels](https://github.com/kubernetes/kubernetes/pull/80238)
+that made the old behavior still achievable. The new alpha feature gate `LegacyNodeRoleBehavior`
+disables the old check, which would allow service load balancers to target masters.
+The risk of regression is low because the code path is simple, and all usage of
+the gate would be internal. We would enable this gate by default in 4.3.
+
+The IPI and UPI install and documentation must ensure that service load balancers
+can correctly target master machines if a 3-node cluster is created by ensuring
+security groups are correctly targeted. It may be desirable to switch the default
+ingress port policy to be NodePort to remove the need to listen on the host in
```

Comment:
I'm confused as to how NodePort can "remove the need to listen on the host" compared to a LoadBalancer which itself uses a NodePort. Help?

Reply:
NodePort vs HostNetwork- just clarifying that we want SLB to be NP by default